src/odemis/driver/tescan.py

```diff
@@ -1127,12 +1160,25 @@ def _doMoveAbs(self, pos):
             logging.debug("Stage move to %s completed", pos)
             self._updatePosition()
 
-    def _doMoveRel(self, shift):
-        # Compute the position in absolute coordinates
+    def _doMoveRel(self, shift: dict):
+        """
+        Request the position in absolute coordinates and create new positions after applying shift values
+        :param shift: shift of linear axes in m or rotational axes in radians
+        """
         self._updatePosition()
         pos = self._position.copy()
+
+        # add the requested change to the current position
         for axis, change in shift.items():
             pos[axis] += change
+            if axis in ("rx", "rz"):
+                rng = self.axes[axis].range
+                # check if any of the range values are 360°, range values for full rotational axis
+                # requested with command StgGetLimits() always returns (-360, 360)
+                if (2 * math.pi) in [abs(ele) for ele in rng]:
+                    # this check is for full rotational axes to handle under- and overflow
+                    if util.almost_equal(abs(rng[0]) - abs(rng[1]), 0.0):
+                        pos[axis] = (pos[axis] - abs(rng[0])) % (2 * math.pi)
```

Comment:
This is odd code. It seems the 2 `if`'s are checking for the same thing: whether the rotational axis supports full rotation or not. I'd suggest to just write the equivalent of "if the rotation range is >= 360°, it's considered to support full rotation". You can write it this way: `if abs(rng[1] - rng[0]) > 2 * math.pi`